atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -155,7 +156,8 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
 
     @Override
     public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition) {
-        StartTransactionWithWatchesResponse responseWithWatches = timelockService.startTransactionWithWatches();
+        StartTransactionWithWatchesResponse responseWithWatches = timelockService
+                .startTransactionWithWatches(OptionalLong.empty());
```

Comment:
I don't think TimelockService should gain a method that's related to lock watches - rather we should be maintaining this state as some part of the hierarchy _below_ timelock service.